federation/pkg/kubefed/unjoin.go

```diff
@@ -81,6 +81,16 @@ func unjoinFederation(f cmdutil.Factory, cmdOut, cmdErr io.Writer, config util.A
 		return nil
 	}
 
+	//we need to ensure deleting the config map created in the deregistered cluster
+	//this configmap was created when the cluster joined this federation to aid
+	//the kube-dns of that cluser to aid service discovery
+	unjoinClusterFactory := config.HostFactory(unjoinFlags.Name, unjoinFlags.Kubeconfig)
```

Comment:
that would mean that the CLUSTER_NAME provided on command line to unjoin was wrong
kubefed unjoin CLUSTER_NAME --host-cluster-context=HOST_CONTEXT [options]
If that cluster context does not exist in this config, we will get an error returned while creating a clientset in deleteConfigMap()
clientset, err := unjoinClusterFactory.ClientSet()
	if err != nil {
		return err
	}
In case the name specified is of some other clusters context then the one joined to this federation
then popCluster() would err and we wouldnt reach here.
I might be missing something, Please suggest!